src/openMVG/matching/regions_matcher.hpp

```diff
@@ -46,100 +47,59 @@ void DistanceRatioMatch
  * as a "database". The other region is then a query that is matched with respect
  * to the database.
  */
-class RegionsMatcher
+class IRegionsMatcher
 {
-  public:
-    
-   /**
-    * @brief The destructor.
-    */ 
-   ~RegionsMatcher() {}
-
-
-    /**
-     * @brief Initialize the matcher by setting one region as a "database".
-     * 
-     * @param[in] regions The Regions to be used as reference when matching another Region.
-     */
-    virtual void Init_database(const features::Regions& regions) = 0;
+public:
+  const features::Regions* regions_ = nullptr;
```

Comment:
i imagine that this cannot be a `const features::Regions& regions_;` without a big cascade mess
But in theory could it be?